src/qrcode/QRCode.tsx

```diff
@@ -15,19 +15,23 @@ import { currentAccountSelector } from 'src/web3/selectors'
 
 interface Props {
   qrSvgRef: React.MutableRefObject<SVG>
+  dataType: QRCodeDataType
 }
 
-const mapStateToProps = (state: RootState): Partial<UriData> => ({
+const mapStateToProps = (state: RootState) => ({
   address: currentAccountSelector(state)!,
   displayName: nameSelector(state) || undefined,
   e164PhoneNumber: state.account.e164PhoneNumber || undefined,
 })
 
-export default function QRCodeDisplay({ qrSvgRef }: Props) {
+export default function QRCodeDisplay({ qrSvgRef, dataType }: Props) {
   const data = useSelector(mapStateToProps, shallowEqual)
+
+  // ValoraDeepLink generates a QR code that deeplinks into the walletconnect send flow of the valora app
+  // Address generates a QR code that has the walletAddress as plaintext that is readable by wallets such as Coinbase and Metamask
   const qrContent = useMemo(
-    () => urlFromUriData(data),
-    [data.address, data.displayName, data.e164PhoneNumber]
+    () => (dataType === QRCodeDataType.ValoraDeepLink ? urlFromUriData(data) : data.address),
```

Comment:
I used a qr code scanner to see what data is in the MetaMask QR code and found this: `ethereum:0x589865Bdd130C77C28B9906DC630bC4349820AEa@42220` which looks like [EIP-681](https://eips.ethereum.org/EIPS/eip-681)

Reply:
Coinbase Wallet uses just the address. However, I think Coinbase can scan EIP-681 contents

Reply:
looks like we could use this https://www.npmjs.com/package/eth-url-parser

Reply:
I think EIP681 sets us up a bit better for cross chain, though supposedly some [bugs have resulted from other wallets not scanning it correctly](https://www.coindesk.com/tech/2021/03/25/coinbase-and-zengo-spar-over-qr-code-standards-that-could-strand-erc-20-tokens/). I don't really care, though

Reply:
well I guess I would care if some centralized exchange parsed a QR code incorrectly and ended up sending funds to a contract address instead of the user's address

not as worried about other wallets, though

Reply:
Forgot to link these discussions together https://valora-app.slack.com/archives/C02KBT0DAHJ/p1671051933339429

Reply:
Jacob replied to this [here](https://valora-app.slack.com/archives/C02898GN22V/p1671057639473929?thread_ts=1671051997.691509&cid=C02898GN22V)